lib/src/native/realm_core.dart

```diff
@@ -764,8 +764,35 @@ class _RealmCore {
     _realmLib.invokeGetBool(() => _realmLib.realm_rollback(realm.handle._pointer), "Could not rollback write");
   }
 
-  void realmRefresh(Realm realm) {
+  bool realmRefresh(Realm realm) {
     _realmLib.invokeGetBool(() => _realmLib.realm_refresh(realm.handle._pointer), "Could not refresh");
+    //TODO: Remove this and return the correct result of calling `realm_refresh`.
+    //Currently there is a bug in C-API that does not return the refresh result correctly.
+    //Issue https://github.com/realm/realm-core/pull/6068
```

Comment:
Should we await this fix before merging?

Reply:
It seems we will anyway, but merging this without it also will work